### ISSUE TITLE
pull request #778 breaks streaming inputs

### ISSUE THREAD

**Issue by fgregg** [wireservice/agate#779]
PR #778 introduced a [`seek`](https://github.com/wireservice/agate/pull/778/files#diff-fcc63e6d6ae4760e758eebd01e78434165cdcb3f5c0baf4532b9531e45faa01bR71) that is not compatible with stdin and other inputs that do not provide seek.

**Comment by fgregg** [comment]
actually not that exact commit, but the changes of yesterday

**Comment by fgregg** [comment]
actually, thie https://github.com/wireservice/agate/pull/778/files#diff-fcc63e6d6ae4760e758eebd01e78434165cdcb3f5c0baf4532b9531e45faa01bR71

**Comment by jpmckinney** [comment]
Did 1.10.1 fix it? Can you provide a way to reproduce?

**Comment by jpmckinney** [comment]
Aha you edited - I’ll add a test.

**Comment by fgregg** [comment]
if you are mucking about in there, i'm pretty sure that you will need a seek or other way of resetting the file in this block.

https://github.com/wireservice/agate/blob/2898d17295cc8a85bafc37942de16def19da1415/agate/table/from_csv.py#L67

**Comment by fgregg** [comment]
not an agate test but, here's a csvkit repro.

```console
echo "a" | csvsort
```

**Comment by jpmckinney** [comment]
Yeah, It can read from stdin `csvsort < examples/dummy.csv`, but not from pipes. Not sure how to test pipes in pytest, so I'll add some sample commands to CI.

```
 if sniff_limit is None: 
     # Reads to the end of the tile, but avoid reading the file twice. 
     handle = StringIO(f.read()) 
     kwargs['dialect'] = csv.Sniffer().sniff(handle.getvalue())
```

We're passing `handle.getvalue()` which is an `str`. The position of the original StringIO doesn't move. Right?

**Comment by fgregg** [comment]
got it. got confused with the overwriting of `handle`

**Comment by jpmckinney** [comment]
I started https://github.com/wireservice/agate/pull/780

The docstring says sniff_limit reads bytes, which is true for stdin (now), but it actually reads characters for the original behavior. I don't know if it's worth forcing reading characters in both cases. Would need something like:

https://stackoverflow.com/a/14292897/244258
https://github.com/racitup/magicrescue-vmdk/blob/master/vmdk.py

I should also see what happens if `sys.stdin.buffer` is passed in.

**Comment by jpmckinney** [comment]
Thank you for reporting! Should be fixed in 1.10.2.
